data/archipelago/scripts/ap_utils.lua

```diff
@@ -235,7 +235,12 @@ function create_our_item_entity(item, x, y)
     	return perk_spawn(x, y, item.perk, true)
   	elseif item.items ~= nil and #item.items > 0 then
     	-- our item is something else (random choice)
-    	return EntityLoad(item.items[Random(1, #item.items)], x, y)
+		local entity_id = EntityLoad(item.items[Random(1, #item.items)], x, y)
+		if item.gold_amount ~= 0 then
+			local life_comp = EntityGetFirstComponent(entity_id, "LifetimeComponent", "enabled_in_world")
+			EntityRemoveComponent(entity_id, life_comp)
+		end
```

Comment:
```suggestion
		local life_comp = EntityGetFirstComponent(entity_id, "LifetimeComponent", "enabled_in_world")
		if life_comp ~= nil then
			EntityRemoveComponent(entity_id, life_comp)
		end
```

May be a more general way to write it.